src/services/formatting/formatting.ts

```diff
@@ -663,9 +663,9 @@ namespace ts.formatting {
                     undecoratedChildStartLine = sourceFile.getLineAndCharacterOfPosition(getNonDecoratorTokenPosOfNode(child, sourceFile)).line;
                 }
 
-                // if child is a list item - try to get its indentation
+                // if child is a list item - try to get its indentation, only if parent is within the original range.
                 let childIndentationAmount = Constants.Unknown;
-                if (isListItem) {
+                if (isListItem && parent.pos >= originalRange.pos && parent.end <= originalRange.end) {

```

Comment:
We've got a `rangeContainsRange` helper function

Reply:
Updated.